substrate/frame/nis/src/benchmarking.rs

```diff
@@ -49,200 +39,276 @@ fn fill_queues<T: Config>() -> Result<(), DispatchError> {
-		Nis::<T>::process_queues(Perquintill::one(), 1, 2, &mut WeightCounter::unlimited());
+		T::Currency::set_balance(
+			&whale,
+			T::ThawThrottle::get()
+				.0
+				.saturating_reciprocal_mul_ceil(T::Currency::balance(&caller)),
+		);
+		Pallet::<T>::place_bid(RawOrigin::Signed(caller.clone()).into(), bid, 1)?;
+		Pallet::<T>::place_bid(RawOrigin::Signed(caller.clone()).into(), bid, 1)?;
+		Pallet::<T>::process_queues(Perquintill::one(), 1, 2, &mut WeightCounter::unlimited());
 		frame_system::Pallet::<T>::set_block_number(Receipts::<T>::get(0).unwrap().expiry);
-		Nis::<T>::communify(RawOrigin::Signed(caller.clone()).into(), 0)?;
-	}: _(RawOrigin::Signed(caller.clone()), 0)
-	verify {
+		Pallet::<T>::communify(RawOrigin::Signed(caller.clone()).into(), 0)?;
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(caller.clone()), 0);
+
 		assert!(Receipts::<T>::get(0).is_none());
+
+		Ok(())
 	}
 
-	process_queues {
+	#[benchmark]
+	fn process_queues() -> Result<(), BenchmarkError> {
 		fill_queues::<T>()?;
-	}: {
-		Nis::<T>::process_queues(
-			Perquintill::one(),
-			Zero::zero(),
-			u32::max_value(),
-			&mut WeightCounter::unlimited(),
-		)
+
+		#[block]
+		{
+			Pallet::<T>::process_queues(
+				Perquintill::one(),
+				Zero::zero(),
+				u32::max_value(),
+				&mut WeightCounter::unlimited(),
+			);
+		}
+
+		Ok(())
 	}
 
-	process_queue {
-		let our_account = Nis::<T>::account_id();
-		let issuance = Nis::<T>::issuance();
+	#[benchmark]
+	fn process_queue() {
+		let our_account = Pallet::<T>::account_id();
+		let issuance = Pallet::<T>::issuance();
 		let mut summary = Summary::<T>::get();
-	}: {
-		Nis::<T>::process_queue(
-			1u32,
-			1u32.into(),
-			&our_account,
-			&issuance,
-			0,
-			&mut Bounded::max_value(),
-			&mut (T::MaxQueueLen::get(), Bounded::max_value()),
-			&mut summary,
-			&mut WeightCounter::unlimited(),
-		)
+
+		#[block]
+		{
+			Pallet::<T>::process_queue(
+				1_u32,
+				1_u32.into(),
+				&our_account,
+				&issuance,
+				0,
+				&mut Bounded::max_value(),
+				&mut (T::MaxQueueLen::get(), Bounded::max_value()),
+				&mut summary,
+				&mut WeightCounter::unlimited(),
+			);
+		}
 	}
 
-	process_bid {
+	#[benchmark]
+	fn process_bid() {
 		let who = account::<T::AccountId>("bidder", 0, SEED);
 		let min_bid = T::MinBid::get().max(One::one());
 		let ed = T::Currency::minimum_balance();
 		T::Currency::set_balance(&who, ed + min_bid);
-		let bid = Bid {
-			amount: T::MinBid::get(),
-			who,
-		};
-		let our_account = Nis::<T>::account_id();
-		let issuance = Nis::<T>::issuance();
+		let bid = Bid { amount: T::MinBid::get(), who };
+		let our_account = Pallet::<T>::account_id();
+		let issuance = Pallet::<T>::issuance();
 		let mut summary = Summary::<T>::get();
-	}: {
-		Nis::<T>::process_bid(
-			bid,
-			2u32.into(),
-			&our_account,
-			&issuance,
-			&mut Bounded::max_value(),
-			&mut Bounded::max_value(),
-			&mut summary,
-		)
+
+		#[block]
+		{
+			Pallet::<T>::process_bid(
```

Comment:
Can we use an `extrinsic_call` here instead of block?

Reply:
`process_bid` is an internal fn.